pkg/ingester/stream_rate_calculator.go

```diff
@@ -0,0 +1,89 @@
+package ingester
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	// defaultStripeSize is the default number of entries to allocate in the
+	// stripeSeries list.
+	defaultStripeSize = 1 << 15
```

Comment:
that's ~32k. Doesn't that mean we're guaranteed to have a conflict if the number of streams is higher than 32k? If so, I feel like 32k is a low value. WDYT?